crates/core/src/db/relational_db.rs

```diff
@@ -42,7 +40,7 @@ pub const ST_TABLE_ID_START: TableId = TableId(2);
 pub struct RelationalDB {
     // TODO(cloutiertyler): This should not be public
     pub(crate) inner: Locking,
-    commit_log: CommitLog,
+    commit_log: Option<CommitLogMut>,
```

Comment:
No it doesn't, the types are laid out a bit confused currently. 

If the storage is configured using `Storage::Memory`, we make it so the `CommitLog` contains no `MessageLog`, but it _always_ contains an `ObjectDB` (which is pointless, because we don't actually access it).

In this patch, I'm dropping the `ObjectDB` if it is not needed, but really it should be in the same `Option` that contains the `MessageLog`. This quickly derails into a large refactoring, so I'll submit that later.